AudioKit/Common/Internals/Audio File/AKAudioFile+Utilities.swift

```diff
@@ -38,63 +38,10 @@ extension AKAudioFile {
         return try AKAudioFile(forReading: silentFile.url)
     }
 
-    /// Returns the time of the peak of the buffer
-    /// - Parameters:
-    ///   - pcmBuffer: A valid AVAudioPCMBuffer
-    /// - Returns: The time in seconds or 0 if it failed
-    static public func findPeak( pcmBuffer: AVAudioPCMBuffer ) -> Double {
-        guard pcmBuffer.frameLength > 0 else { return 0 }
-        guard let floatData = pcmBuffer.floatChannelData else { return 0 }
-
-        var framePosition = 0
-        var position = 0
-        var lastPeak: Float = -10_000.0
-        let frameLength = 512
-        let channelCount = Int(pcmBuffer.format.channelCount)
-
-        while true {
-            if position + frameLength >= pcmBuffer.frameLength {
-                break
-            }
-            for channel in 0 ..< channelCount {
-                var block = Array(repeating: Float(0), count: frameLength)
-
-                // fill the block with frameLength samples
-                for i in 0 ..< block.count {
-                    if i + position >= pcmBuffer.frameLength {
-                        break
-                    }
-                    block[i] = floatData[channel][i + position]
-                }
-                // scan the block
-                let peak = AKAudioFile.getPeak(from: block)
-
-                if peak > lastPeak {
-                    framePosition = position
-                    lastPeak = peak
-                }
-                position += block.count
-            }
-        }
-
-        let time = Double(framePosition / pcmBuffer.format.sampleRate)
-        return time
+    @available(*, deprecated, renamed: "AKPlayer.findPeak(pcmBuffer:)")
```

Comment:
This seems backwards to me.  The peak finding should be at the file or buffer level, then the player should call into that.

Reply:
I don't to use AKAudioFile any longer, but if you think it should stay around I'll move this back. This is my function btw.

Reply:
Basically I was wondering if it would be better to have a utility class for these kinds of useful but static things.

Reply:
If I were to implement it myself, I would probably do an extension on AVAudioPCMBuffer because that is common denominator.
```let floatData = pcmBuffer.floatChannelData```

Reply:
It would look like this ```let peak = myBuffer.peak()```

Reply:
Or let peak = myBuffer.floatPeak(), since buffers can contain non-float data.

Reply:
yeah, i considered that. i haven't entirely liked the blurring of lines between extensions and foundation classes. I suppose it happens a lot in other places in AudioKit, so I think I'll take your advice. Where are these type of extensions going?

Reply:
AudioKitHelpers?

Reply:
AudioKit/AudioKit/Common/Internals/Utilities/AVAudioBufferConvenience.swift

Reply:
AVAudioBufferConvenience.swift ?

Reply:
went with peakTime as a property.